clang/test/SemaHLSL/BuiltIns/half-float-only-errors2.hlsl

```diff
@@ -0,0 +1,7 @@
+// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm-only -disable-llvm-passes -verify -DTEST_FUNC=__builtin_elementwise_atan2
+// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm-only -disable-llvm-passes -verify -DTEST_FUNC=__builtin_elementwise_pow
+

```

Comment:
add 
```suggestion
// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm-only -disable-llvm-passes -verify -DTEST_FUNC=__builtin_elementwise_fmod
```
delete the  https://github.com/llvm/llvm-project/blob/main/clang/test/SemaHLSL/BuiltIns/fmod-errors.hlsl file

Reply:
I can do this, but `fmod-errors.hlsl` tests a few extra types (`bool`, `int2`, scalar `double`).  You suggest adding the scalar double case below, but shouldn't we do that for `half-float-only-errors.hlsl` as well?  Then, should we add `bool` and `int` tests to both as well?

Furthermore, I noticed that with the `bool` argument (from fmod test), the error is:
> 1st argument must be a vector, integer or floating point type (was 'bool')

`integer` should not be suggested here as a valid type, so I think there is another fix to be made there.

Reply:
the only HLSL specific checks are to exclude double the rest of those tests have nothing to do with SemaHLSL

Reply:
we can update `half-float-only-errors.hlsl` in a different NFC PR.

Reply:
Sorry, I updated that one here.  Is that ok?